toolkit/tools/scheduler/scheduler.go

```diff
@@ -200,7 +201,7 @@ func cancelBuildsOnSignal(signals chan os.Signal, agent buildagents.BuildAgent)
 
 // buildGraph builds all packages in the dependency graph requested.
 // It will save the resulting graph to outputFile.
-func buildGraph(inputFile, outputFile string, agent buildagents.BuildAgent, workers, buildAttempts int, checkAttempts int, stopOnFailure, canUseCache bool, packagesToBuild []*pkgjson.PackageVer, packagesNamesToRebuild, ignoredPackages, reservedFiles []string, deltaBuild bool) (err error) {
+func buildGraph(inputFile, outputFile string, agent buildagents.BuildAgent, workers, buildAttempts int, checkAttempts int, stopOnFailure, canUseCache bool, packagesToBuild []*pkgjson.PackageVer, packagesNamesToRebuild, ignoredPackages, reservedFiles []string, deltaBuild bool, supressConflictingRPMs bool) (err error) {
```

Comment:
A note on API design/ergonomics/beauty; IMHO for 3.0 we should take a look at these details where our function API has grown out of bounds due to gradual changes. I think we should not have functions that take more than certain number of arguments (may be more than 4? -- debatable).

We should probably look at a cleaner way of doing this as I believe things will change and we may need to add/drop a few of the stuff. Our APIs should be such that, we can easily add/modify new stuff and still not have to do a mass refactor just to propagate a variable.

e.g one idea could be:

```go
struct BuildGraphConfig {
          inputGraphFile: string,
          outputGraphFile: string,
          //...
}

func (bc *BuildGraphConfig) buildGraph(packagesToBuild []*pkgjson.PackageVer) (error) {
        // process
}
```

One idea of above is our `intent` to build packages, rest all IMO is not too important for the reader.

I also really like the idea of Functional options: [Functional options for friendly APIs](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis)

Although, this may not be the best place for a comment like this :(, its just an idea I have been thinking on from sometime.

Reply:
That sounds like a great idea we should keep and maybe add to backlog?

Reply:
Agreed, its something I've been wishing I had the time to fix for a while (especially looking at you `imager.go`...)